plugins/de.cognicrypt.core/src/de/cognicrypt/utils/Utils.java

```diff
@@ -403,9 +403,14 @@ public static int getFirstIndexofUCL(final String searchString) {
 	 * @throws MalformedURLException
 	 */
 	public static CryptSLRule getCryptSLRule(String cryptslRule) throws MalformedURLException {
-		File ruleRes = Utils.getResourceFromWithin(Constants.RELATIVE_CUSTOM_RULES_DIR + "/" + cryptslRule + RuleFormat.SOURCE.toString(), de.cognicrypt.core.Activator.PLUGIN_ID);
+		String defaultRuleset = Constants.Rules.JavaCryptographicArchitecture.toString();
+		String[] versionRules = getRuleVersions(defaultRuleset);
+		String latestVersion = versionRules[versionRules.length - 1];
+		String slash = File.separator;
+		String srcFormat = RuleFormat.SOURCE.toString();
+		File ruleRes = new File(System.getProperty("user.dir") + slash + defaultRuleset + slash + latestVersion + slash + defaultRuleset + slash + cryptslRule + srcFormat);

```

Comment:
Here, please use Constants.ECLIPSE_RULES_DIR.

Reply:
Also in-line all objects that are only used once (like defaultRuleset, latestVersion, etc.